storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -1096,6 +1113,20 @@ void Redo_Log_Data_Manager::copy_func() {
       return;
     }
 
+    if (xtrabackup_register_redo_log_consumer) {
+      if (archived_log_monitor.is_ready() &&
+          archived_log_state == ARCHIVED_LOG_POSITIONED) {
+        redo_log_consumer.deinit(redo_log_consumer_cnx);
+        mysql_close(redo_log_consumer_cnx);
+        xtrabackup_register_redo_log_consumer = false;
+      } else {
+        if (consumer_lsn != reader.get_scanned_lsn()) {
+          consumer_lsn = reader.get_scanned_lsn();
+          redo_log_consumer.advance(redo_log_consumer_cnx, consumer_lsn);
```

Comment:
we are already advancing above in the copy_func(). Can you please explain why do we need this and possibly add as comment?

how do we deal with archive log that is not yet positioned to desired LSN but has started scanning and doing skip_for_block(LSN)?

Reply:
@altmannmarcelo

Reply:
@satya-bodapati on copy_func we advance consumer twice:

* once we start the thread (where we basically set the lower boundary to checkpoint LSN as we did not start to copy anything yet.
* inside the while loop that continues to copy redo log blocks to advance the lower boundary to the last copied lsn from copy_once

> how do we deal with archive log that is not yet positioned to desired LSN but has started scanning and doing skip_for_block(LSN)?

While we have not fully switched to archive ( archived_log_monitor.is_ready() is false & archived_log_state != ARCHIVED_LOG_POSITIONED) we still hold the lower boundary to the last copied lsn from reader.get_scanned_lsn(). Once we fully switched to archive log, we deinit redo log consumer as its safe to let the server purge redo log now as archive component will itself be registered as a redo log consumer.

Reply:
Thanks, I was a bit confused with ARCHIVED_LOG_MATCHED. POSITIONED ensured that we have found the desired LSN in the archive redo file.